docs/aws/audit/snsmonitoring/rules/is_topic_exposed.mdx

```diff
@@ -23,6 +23,229 @@ HITRUST, AWSWAF, SOC2, NISTCSF, PCIDSS
+   - Save the changes to the access policy.
+
+By following these steps, you can ensure that your SNS topics are not exposed to unauthorized access.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent SNS Topics from being exposed in AWS using the AWS CLI, you can follow these steps:
+
+1. **List SNS Topics**:
+   First, identify the SNS topics in your account.
+   ```sh
+   aws sns list-topics
+   ```
+
+2. **Get Topic Attributes**:
+   For each topic, retrieve its attributes to check the current policy.
+   ```sh
+   aws sns get-topic-attributes --topic-arn <topic-arn>
+   ```
+
+3. **Update Topic Policy**:
+   Modify the topic policy to restrict access. Ensure that only specific AWS accounts or IAM roles/users have access. Here is an example of a restrictive policy:
+   ```sh
+   aws sns set-topic-attributes --topic-arn <topic-arn> --attribute-name Policy --attribute-value '{
+     "Version": "2012-10-17",
+     "Statement": [
+       {
+         "Effect": "Allow",
+         "Principal": {
+           "AWS": "arn:aws:iam::<account-id>:root"
+         },
+         "Action": "SNS:Publish",
+         "Resource": "<topic-arn>"
+       }
+     ]
+   }'
+   ```
+
+4. **Verify Policy Update**:
+   Confirm that the policy has been updated correctly by retrieving the topic attributes again.
+   ```sh
+   aws sns get-topic-attributes --topic-arn <topic-arn>
+   ```
+
+By following these steps, you can ensure that your SNS topics are not exposed and are only accessible by authorized entities.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent SNS Topics from being exposed in AWS using Python scripts, you can use the Boto3 library, which is the AWS SDK for Python. Here are the steps to ensure that your SNS topics are not exposed:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed. If not, you can install it using pip.
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session with your AWS credentials and region.
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+   sns_client = session.client('sns')
+   ```
+
+3. **List SNS Topics**:
+   Retrieve the list of all SNS topics in your account.
+   ```python
+   response = sns_client.list_topics()
+   topics = response['Topics']
+   ```
+
+4. **Check and Update SNS Topic Policies**:
+   Iterate through each topic and ensure that the policy does not allow public access. If it does, update the policy to restrict access.
+   ```python
+   import json
+
+   for topic in topics:
+       topic_arn = topic['TopicArn']
+       policy_response = sns_client.get_topic_attributes(TopicArn=topic_arn)
+       policy = json.loads(policy_response['Attributes']['Policy'])
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>